### PR TITLE
doc: Temporal Filters

### PR DESCRIPTION
**Pull request by msudberg**
* Break out Guides section for task explanations that don't quite fit into our current doc structure
* Create stub temporal filters page
* Move "Idioms" (which only contains top k) to a "Top K" page
* Add restrictions that apply to mz_logical_timestamp() (See #7164)

### CODE REVIEW

File: doc/user/content/guides/top-k.md [hunk 1]
```diff
@@ -1,21 +1,13 @@
 ---
-title: "Idioms"
-description: "Recommended SQL idioms."
-weight: 12
+title: "Top K by Group"
+description: "Filter out all but the top elements in a group."
```
Comment (morsapaes) [inline doc/user/content/guides/top-k.md:3]: Could "Find the top- or bottom-K elements in a group." be a simpler and more straightforward definition of Top K? Whenever I've used this function, my goal was always to **find** the top or bottom K elements, so describing it as "filter out all but the top elements" reads a bit weird.

**Comment by awang** [comment]
next steps: add a brief summary of temporal filters at /guides/temporal-filters/ and then link out to the blog post for more info

**Review by frankmcsherry — APPROVED** [review]
Looks good to me. Thanks!

File: doc/user/content/sql/functions/now_and_mz_logical_timestamp.md [hunk 2]
```diff
@@ -19,64 +19,24 @@ In Materialize, `now()` doesn't represent the system time, as it does in most sy
...
+For more information, see [Temporal Filters: Enabling Windowed Queries in Materialize](https://materialize.com/temporal-filters/).
 
-### Temporal filter using mz_logical_timestamp()
+## Restrictions
 
-For this example, you'll need to create a sample data source and create a materialized view from it for later reference.
+You can only use `mz_logical_timestamp()` to establish a temporal filter in one of the following types of clauses:
 
-```sql
---Create table
```
Comment (awang) [inline doc/user/content/sql/functions/now_and_mz_logical_timestamp.md:31]: nit: the 2 sql comments seem a bit unnecessary

**Review by awang — APPROVED** [review]
cool!

File: doc/user/content/sql/functions/now_and_mz_logical_timestamp.md (hunk 2 above)
Comment (msudberg) [reply doc/user/content/sql/functions/now_and_mz_logical_timestamp.md:31]: I think it's helpful to break up the big chunks of code.

**Pull request merged**